mlflow/prompt/registry_utils.py

```diff
@@ -43,11 +46,23 @@ def is_prompt_supported_registry(registry_uri: Optional[str] = None) -> bool:
     """
     Check if the current registry supports prompts.
 
-    Prompts registration is supported only in the OSS MLflow Tracking Server,
-    not in Databricks or OSS Unity Catalog.
+    Prompts registration is supported in:
+    - OSS MLflow Tracking Server (always)
+    - Unity Catalog (when MLFLOW_ENABLE_UC_PROMPT_SUPPORT is enabled)
+    - Not supported in legacy Databricks workspace registry
     """
     registry_uri = registry_uri or mlflow.get_registry_uri()
-    return not registry_uri.startswith("databricks") and not registry_uri.startswith("uc:")
+    
+    # Legacy Databricks workspace registry doesn't support prompts
+    if registry_uri.startswith("databricks") and not registry_uri.startswith("databricks-uc"):
+        return False
+    
+    # UC registries support prompts when feature flag is enabled
+    if registry_uri.startswith("databricks-uc") or registry_uri.startswith("uc:"):
```

Comment:
@rohitarun-db Are you sure we should be checking "uc:/"? Does this correspond to OSS UC, which doesn't support prompt registry yet?

cc @kriscon-db